sources/packages/api/src/services/application/application.service.ts

```diff
@@ -675,35 +684,154 @@ export class ApplicationService extends RecordDataModelService<Application> {
+      .leftJoin(
+        "currentAssessment.studentScholasticStanding",
+        "studentScholasticStanding",
+      )
       .innerJoin("application.student", "student")
       .innerJoin("student.user", "user")
       .where("application.location.id = :locationId", { locationId })
       .andWhere("application.applicationStatus = :applicationStatus", {
         applicationStatus: ApplicationStatus.completed,
-      })
-      .orderBy("application.applicationNumber", "DESC")
-      .getMany();
+      });
+
+    // Available applications are records that have no 'scholastic standing reported' and are not 'archived'.
+    if (applicationStatus === ApplicationStatus.available) {
+      activeApplicationQuery.andWhere(
+        new Brackets((qb) => {
+          qb.where("studentScholasticStanding.id IS NULL").andWhere(
+            "application.isArchived = :isArchived",
+            {
+              isArchived: false,
+            },
+          );
+        }),
+      );
+    }
+
+    // Unavailable applications are records that are either 'completed (scholastic standing reported but not yet archived)' or 'archived'.
+    if (applicationStatus === ApplicationStatus.unavailable) {
+      activeApplicationQuery.andWhere(
+        new Brackets((qb) => {
+          qb.where("studentScholasticStanding.id IS NOT NULL").andWhere(
+            "application.isArchived = :isNotArchived",
+            {
+              isNotArchived: false,
+            },
+          );
+        }),
+      );
+      activeApplicationQuery.orWhere(
+        new Brackets((qb) => {
+          qb.where("application.isArchived = :isArchived", {
+            isArchived: true,
+          });
+        }),
+      );
+    }
+    activeApplicationQuery.orderBy("application.applicationNumber", "DESC");
+
+    if (paginationOptions.searchCriteria) {
+      activeApplicationQuery
+        .andWhere(
+          new Brackets((qb) => {
+            qb.where(
+              "CONCAT(user.firstName,' ', user.lastName) Ilike :searchCriteria",
+            ).orWhere("application.applicationNumber Ilike :searchCriteria");
+          }),
+        )
+        .setParameter(
+          "searchCriteria",
+          `%${paginationOptions.searchCriteria.trim()}%`,
+        );
+    }
+    activeApplicationQuery
+      .orderBy(
+        this.transformToEntitySortField(
+          paginationOptions.sortField,
+          paginationOptions.sortOrder,
+        ),
+      )
+      .offset(paginationOptions.page * paginationOptions.pageLimit)
+      .limit(paginationOptions.pageLimit);
+    const [result, count] = await activeApplicationQuery.getManyAndCount();
+    return {
+      results: result,
+      count: count,
+    };
+  }
+
+  /**
+   * Returns application status with respect to archive status and scholastic standing change.
+   * @param isArchived archive status.
+   * @param studentScholasticStandingId scholastic standing id.
+   * @returns application status.
+   */
+  getApplicationStatus(
```

Comment:
This method/status seems to be too coupled with the scholastic standing. I would recommend renaming it to something like `getApplicationSholasticStandingStatus` and the enum to be returned can be `ApplicationSholasticStandingStatus`.